src/metal.cs

```diff
@@ -2892,8 +3173,158 @@ interface MTLTileRenderPipelineDescriptor : NSCopying {
+		ulong SignaledValue { get; set; }
+	}
+
+	[Mac (10,14, onlyOn64: true), iOS (12,0), TV (12,0)]
+	[BaseType (typeof(NSObject))]
+	interface MTLSharedEventHandle : NSSecureCoding {
+		[NullAllowed, Export ("label")]
+		string Label { get; }
+	}
+
+	interface IMTLIndirectRenderCommand {}
+
+	[Mac (10,14, onlyOn64: true), iOS (12,0)]
+	[Protocol]
+	interface MTLIndirectRenderCommand {
+		[NoiOS, NoTV, Mac (10,14, onlyOn64: true)]
+		[Abstract]
+		[Export ("setRenderPipelineState:")]
+		void SetRenderPipelineState (IMTLRenderPipelineState pipelineState);
+
+		[Abstract]
+		[Export ("setVertexBuffer:offset:atIndex:")]
+		void SetVertexBuffer (IMTLBuffer buffer, nuint offset, nuint index);
+
+		[Abstract]
+		[Export ("setFragmentBuffer:offset:atIndex:")]
+		void SetFragmentBuffer (IMTLBuffer buffer, nuint offset, nuint index);
+
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[NoiOS, NoTV]
+		[Export ("drawPatches:patchStart:patchCount:patchIndexBuffer:patchIndexBufferOffset:instanceCount:baseInstance:tessellationFactorBuffer:tessellationFactorBufferOffset:tessellationFactorBufferInstanceStride:")]
+		void DrawPatches (nuint numberOfPatchControlPoints, nuint patchStart, nuint patchCount, [NullAllowed] IMTLBuffer patchIndexBuffer, nuint patchIndexBufferOffset, nuint instanceCount, nuint baseInstance, IMTLBuffer buffer, nuint offset, nuint instanceStride);
+
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[NoiOS, NoTV]
+		[Export ("drawIndexedPatches:patchStart:patchCount:patchIndexBuffer:patchIndexBufferOffset:controlPointIndexBuffer:controlPointIndexBufferOffset:instanceCount:baseInstance:tessellationFactorBuffer:tessellationFactorBufferOffset:tessellationFactorBufferInstanceStride:")]
+		void DrawIndexedPatches (nuint numberOfPatchControlPoints, nuint patchStart, nuint patchCount, [NullAllowed] IMTLBuffer patchIndexBuffer, nuint patchIndexBufferOffset, IMTLBuffer controlPointIndexBuffer, nuint controlPointIndexBufferOffset, nuint instanceCount, nuint baseInstance, IMTLBuffer buffer, nuint offset, nuint instanceStride);
+
+		[Abstract]
+		[Export ("drawPrimitives:vertexStart:vertexCount:instanceCount:baseInstance:")]
+		void DrawPrimitives (MTLPrimitiveType primitiveType, nuint vertexStart, nuint vertexCount, nuint instanceCount, nuint baseInstance);
+
+		[Abstract]
+		[Export ("drawIndexedPrimitives:indexCount:indexType:indexBuffer:indexBufferOffset:instanceCount:baseVertex:baseInstance:")]
+		void DrawIndexedPrimitives (MTLPrimitiveType primitiveType, nuint indexCount, MTLIndexType indexType, IMTLBuffer indexBuffer, nuint indexBufferOffset, nuint instanceCount, nint baseVertex, nuint baseInstance);
+
+		[Abstract]
+		[Export ("reset")]
+		void Reset ();
+	}
+
+	[Mac (10,14, onlyOn64: true), iOS (12,0), TV (12,0)]
+	[BaseType (typeof(NSObject))]
+	interface MTLIndirectCommandBufferDescriptor {
+		[Export ("commandTypes", ArgumentSemantic.Assign)]
+		MTLIndirectCommandType CommandTypes { get; set; }
+
+		[NoiOS, NoTV, Mac (10,14, onlyOn64: true)]
```

Comment:
you don't need to duplicate `Mac (10,14, onlyOn64: true)` since it's already on the type